src/extra/utils/ftd.pyx

```diff
@@ -0,0 +1,354 @@
+# cython: boundscheck=False, wraparound=False, cdivision=True
+
+from cython cimport floating
+from cython.view cimport contiguous
+from libc.math cimport M_PI, sin, fabs, floor, ceil
+
+import numpy as np
+
+
+"""Edge interpolation method."""
+cpdef enum EdgeInterpolation:
+    NEAREST = 0
+    LINEAR = 1
+    SPLINE = 2
+    SINC = 3
+
+ctypedef floating data_t
+ctypedef floating[::contiguous] data_array_t
+
+
+cdef inline double _sinc(double x) nogil:
+    """Normalized sinc function."""
+
+    if x != 0.0:
+        return sin(x*M_PI) / (x*M_PI)
+    else:
+        return 1.0
+
+
+def sinc_interpolate(
+    double[::contiguous] y_sampled, double[::contiguous] x_interp,
+    double[::contiguous] y_interp = None, int window=100
+):
+    # TODO: Useful function that should go somewhere, but doesn't
+    # actually belong to this module.
+
+    if y_interp is None:
+        y_interp = np.zeros(len(x_interp), dtype=np.float64)
```

Comment:
If `y_interp` is passed, I think you're assuming it's the same size as `x_interp`, right? `i` is used as an index for both.

As you're (sensibly) disabling bounds checking, I would explicitly check this before the loop. It's a small performance cost, but I think functions you can call from Python code should never segfault just because you made a mistake with the arguments.

Reply:
Good point, missed that one.